### PR TITLE
Hydrator updates EmbedOne instead of replacing (if possible)

### LINKED ISSUE #968: Embedded document and Query::HINT_REFRESH
It looks like refresh queries using ``Query::HINT_REFRESH`` are creating a new embedded document instead of updating values of existing one.

For example:

```php
$embed = $document->getEmbed();
// ... do query with HINT_REFRESH
echo $embed === $document->getEmbed() ? "Same" : "Not";
```
Outputs: **Not** :cry:

### PR DESCRIPTION
**Pull request by malarzm**
Fixes #968 

/cc @vkartaviy

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Hydrator/HydratorFactory.php [hunk 1]
```diff
@@ -319,6 +319,10 @@ private function generateHydratorClass(ClassMetadata $class, $hydratorClassName,
             \$className = \$this->unitOfWork->getClassNameForAssociation(\$this->class->fieldMappings['%2\$s'], \$embeddedDocument);
             \$embeddedMetadata = \$this->dm->getClassMetadata(\$className);
             \$return = \$embeddedMetadata->newInstance();
```
Comment (vkartaviy) [inline lib/Doctrine/ODM/MongoDB/Hydrator/HydratorFactory.php]: It seems the new instance creation should be moved to the ```else``` block of ```instanceof``` check statement.

**Comment by vkartaviy** [comment]
Btw - What about ```EmbedMany``` case?

File: lib/Doctrine/ODM/MongoDB/Hydrator/HydratorFactory.php (hunk 1 above)
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Hydrator/HydratorFactory.php]: Right, always a saving

**Comment by malarzm** [comment]
@vkartaviy I'll see what I can do about ``EmbedMany``

**Comment by malarzm** [comment]
Handling ``EmbedMany`` is far from being trivial and I believe there's no easy way to refresh objects there like proposed with ``EmbedOne``. After discussion with @jwage we think that being consistent is better therefore we're not merging this in

**Pull request closed without merging**